### PR TITLE
Add observabilityclient config to env vars

### PR DESCRIPTION
**Pull request by vyzigold**
This adds the needed variables for the observabilityclient to work. Observabilityclient allows to execute `openstack metric` command for interacting with prometheus deployed as part of telemetry MetricStorage by the telemetry-operator.

I should note that the observabilityclient currently isn't available in the openstackclient container image, so the `openstack metric` commands aren't available at the moment without manually installing it. But the PR itself doesn't require observabilityclient installed on the container.

Depends-On: https://github.com/openstack-k8s-operators/openstack-operator/pull/641

### CODE REVIEW

**Comment by vyzigold** [comment]
Work on readding observabilityclient to the image is ongoing here: https://github.com/openstack-k8s-operators/tcib/pull/122

**Comment by vyzigold** [comment]
I think the issue might be, that the added telemetry-operator dependency is too new and the operators in go.mod need to be updated first. I added a depends-on on that PR. I'll take a look at this once that's merged.

File: hack/pin-bundle-images.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ if [ -n "$DOCKERFILE" ]; then
 fi
 
 #loop over each openstack-k8s-operators go.mod entry
-for MOD_PATH in $(go list -mod=readonly -m -json all | jq -r '. | select(.Path | contains("openstack")) | .Replace // . |.Path' | grep -v openstack-operator | grep -v lib-common); do
+for MOD_PATH in $(go list -mod=readonly -m -json all | jq -r '. | select(.Path | contains("openstack")) | .Replace // . |.Path' | grep -v openstack-operator | grep -v lib-common | grep -v telemetry-operator$); do
```
Comment (abays) [inline hack/pin-bundle-images.sh:20]: Why is this change needed?

Comment (vyzigold) [reply hack/pin-bundle-images.sh:20]: I added telemetry-operator to go.mod (not telemetry-operator/api). This totally confused the rest of the code. This change is there to remove this new dependency. You can see the error here: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openstack-k8s-operators_openstack-operator/643/pull-ci-openstack-k8s-operators-openstack-operator-main-openstack-operator-build-deploy-kuttl/1754392910935101440

**Comment by vyzigold** [comment]
I rebased on top of main

**Comment by vyzigold** [comment]
recheck
```
Failed to add subports to trunk 'f738f107-af28-4a9f-a0c0-773c84b5392f': ConflictException: 409: Client Error for url: https://network.rdo.mtl2.vexxhost.net/v2.0/trunks/f738f107-af28-4a9f-a0c0-773c84b5392f/add_subports, Port 91b881f3-dce8-44c8-b5fc-fa43db8f9083 is in use by another trunk.
```
```
ConflictException: 409: Client Error for url: https://compute.rdo.mtl2.vexxhost.net/v2.1/servers/4f95266a-ae23-4449-b865-04ab0b1c68b1/os-interface, Port 512b2651-37b3-4009-b5b4-c3df39d5b916 is still in use.
```

**Comment by vyzigold** [comment]
recheck
```
ERROR! Unexpected Exception, this is probably a bug: [Errno 117] Structure needs cleaning: 'refs'
```

**Comment by vyzigold** [comment]
recheck

**Comment by vyzigold** [comment]
recheck

**Comment by vyzigold** [comment]
recheck
tempest.scenario.test_network_basic_ops.TestNetworkBasicOps failed even though it succeeded before. This shouldn't be related to this PR anyway.
```
AssertionError: False is not true : Public network connectivity check failed
Timed out waiting for 192.168.122.238 to become reachable
```

**Comment by vyzigold** [comment]
recheck
Different tempest scenarios failed this time:
```
tempest.scenario.test_server_volume_attachment.TestServerVolumeAttachScenarioOldVersion
tempest.scenario.test_server_volume_attachment.TestServerVolumeAttachmentScenario
```

**Comment by vyzigold** [comment]
recheck

**Comment by vyzigold** [comment]
I squashed the commits, rebased on top of main and used the latest telemetry-operator api. This allowed me to drop the dependency on the telemetry-operator.

File: hack/pin-bundle-images.sh (hunk 1 above)
Comment (vyzigold) [reply hack/pin-bundle-images.sh]: This change isn't needed in the latest version anymore and it was removed.

**Comment by vyzigold** [comment]
recheck

**Comment by vyzigold** [comment]
recheck
memchached is trying to use an non-existing image. I saw some tests succeed this morning, so I hope the issue is fixed.
```
  Normal   Pulling         30m (x4 over 31m)    kubelet            Pulling image "quay.io/podified-antelope-centos9/openstack-memcached@sha256:2aa527adcae69681592973e21d18e81a866627f43aed048822defcaf04949582"
  Warning  Failed          30m (x4 over 31m)    kubelet            Failed to pull image "quay.io/podified-antelope-centos9/openstack-memcached@sha256:2aa527adcae69681592973e21d18e81a866627f43aed048822defcaf04949582": rpc error: code = Unknown desc = reading manifest sha256:2aa527adcae69681592973e21d18e81a866627f43aed048822defcaf04949582 in quay.io/podified-antelope-centos9/openstack-memcached: manifest unknown
  Warning  Failed          30m (x4 over 31m)    kubelet            Error: ErrImagePull
  Warning  Failed          29m (x6 over 31m)    kubelet            Error: ImagePullBackOff
```

**Review by abays — APPROVED** [review]
/lgtm
